packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -129,6 +129,34 @@ moduleFor('Syntax test: {{#with as}}', class extends TogglingSyntaxConditionalsT
     this.assertText('No Thing bar');
   }
 
+  ['@test can access alias of a proxy']() {
+    this.render(`{{#with proxyThing as |person|}}{{person.name}}{{/with}}`, {
+      proxyThing: { isTruthy: true, name: 'Tom Dale' }
```

Comment:
Should I use a real proxy here? My concern is that we will remove the actual proxy implementations when the the addons get EOL'ed, and when we do that, we might end up removing this test.

(Actually, are the `isTruthy` stuff intrinsically important, or are they just there to help us implement proxies? i.e. can we remove support for `isTruthy` when we remove proxies?)